#! /bin/sh /usr/share/dpatch/dpatch-run
## rtnl_dump_filter.dpatch by Gaetan Bisson
##
## DP: netlink: Updated the signature of rtnl_dump_filter to recent iproute2.

@DPATCH@

diff -Naur old/src/netlink.c new/src/netlink.c
--- old/src/netlink.c   2012-01-22 22:10:04.000000000 +1100
+++ new/src/netlink.c   2012-02-01 17:22:54.362752065 +1100
@@ -570,8 +570,7 @@
     return (-1);
   }
 
-  if (rtnl_dump_filter (&rth, link_filter, /* arg1 = */ NULL,
-	NULL, NULL) != 0)
+  if (rtnl_dump_filter (&rth, link_filter, /* arg1 = */ NULL) != 0)
   {
     ERROR ("netlink plugin: ir_read: rtnl_dump_filter failed.");
     return (-1);
@@ -608,8 +607,7 @@
 	continue;
       }
 
-      if (rtnl_dump_filter (&rth, qos_filter, (void *) &ifindex,
-	    NULL, NULL) != 0)
+      if (rtnl_dump_filter (&rth, qos_filter, (void *) &ifindex) != 0)
       {
 	ERROR ("netlink plugin: ir_read: rtnl_dump_filter failed.");
 	continue;
